lib/transaction/specialtransactions/payload/subtxtransitionpayload.js

```diff
@@ -0,0 +1,226 @@
+
+  Preconditions.checkArgument(isHexString(blockchainUserPayload.regTxId), 'expect regTxId to be a hex string but got ' + typeof blockchainUserPayload.regTxId);
+  Preconditions.checkArgument(blockchainUserPayload.regTxId.length === constants.SHA256_HASH_SIZE * 2, 'Invalid regTxId size');
+
+  Preconditions.checkArgument(isHexString(blockchainUserPayload.hashPrevSubTx), 'expect hashPrevSubTx to be a hex string but got ' + typeof blockchainUserPayload.hashPrevSubTx);
+  Preconditions.checkArgument(blockchainUserPayload.hashPrevSubTx.length === constants.SHA256_HASH_SIZE * 2, 'Invalid hashPrevSubTx size');
+
+  Preconditions.checkArgument(isHexString(blockchainUserPayload.hashSTPacket), 'expect hashSTPacket to be a hex string but got ' + typeof blockchainUserPayload.hashSTPacket);
+  Preconditions.checkArgument(blockchainUserPayload.hashSTPacket.length === constants.SHA256_HASH_SIZE * 2, 'Invalid hashSTPacket size');
+
+
+  if (blockchainUserPayload.vchSig) {
+    Preconditions.checkArgument(isHexString(blockchainUserPayload.vchSig), 'expect vchSig to be a hex string but got ' + typeof blockchainUserPayload.vchSig);
+    Preconditions.checkArgument(blockchainUserPayload.vchSig.length === constants.COMPACT_SIGNATURE_SIZE * 2, 'Invalid vchSig size');
+  }
+};
+
+/* Instance methods */
+
+/**
+ * Validates payload data
+ * @return {boolean}
+ */
+SubTxTransitionPayload.prototype.validate = function() {
+  return SubTxTransitionPayload.validatePayloadJSON(this.toJSON());
+};
+
+/**
+ * @param {string} regTxId - Hex string
+ */
+SubTxTransitionPayload.prototype.setRegTxId = function(regTxId) {
+  this.regTxId = regTxId;
+  return this;
+};
+
+/**
+ * @param {string} hashPrevSubTx - Hex string
+ * @return {SubTxTransitionPayload}
+ */
+SubTxTransitionPayload.prototype.setHashPrevSubTx = function(hashPrevSubTx) {
+  this.hashPrevSubTx = hashPrevSubTx;
+  return this;
+};
+
+/**
+ * @param {string} hashSTPacket - Hex string
+ * @return {SubTxTransitionPayload}
+ */
+SubTxTransitionPayload.prototype.setHashSTPacket = function(hashSTPacket) {
+  this.hashSTPacket = hashSTPacket;
+  return this;
+};
+
+/**
+ * @param {number} creditFee
+ * @return {SubTxTransitionPayload}
+ */
+SubTxTransitionPayload.prototype.setCreditFee = function(creditFee) {
+  this.creditFee = creditFee;
+  return this;
+};
+
+/**
+ * Serializes payload to JSON
+ * @param [options]
+ * @param {boolean} options.skipSignature - skip signature part. Needed for creating new signature
+ * @return {TransitionPayloadJSON}
+ */
+SubTxTransitionPayload.prototype.toJSON = function toJSON(options) {
+  var includeSignature = !options || (options && !options.skipSignature);
+  var payloadJSON = {
+    nVersion: this.nVersion,
+    regTxId: this.regTxId,
+    hashPrevSubTx: this.hashPrevSubTx,
+    creditFee: this.creditFee,
+    hashSTPacket: this.hashSTPacket
+  };
+  if (includeSignature && !!this.vchSig) {
```

Comment:
Guys from Airbnb suggest to use `Boolean(...)`. More obvious.

Reply:
Fixed!